libyang/context.py

```diff
@@ -424,7 +440,33 @@ def parse_data_file(
             fmt,
             in_type=IOType.FD,
             in_data=fileobj,
-            parent=parent,
+            parser_lyb_mod_update=parser_lyb_mod_update,
+            parser_no_state=parser_no_state,
+            parser_parse_only=parser_parse_only,
+            parser_opaq=parser_opaq,
+            parser_ordered=parser_ordered,
+            parser_strict=parser_strict,
+            validation_no_state=validation_no_state,
+            validation_validate_present=validation_validate_present,
+        )
```

Comment:
This should call `lyd_parse_data_fd` directly. I don't see the point in wrapping the call into a generic parse_data that ends up calling specific functions.